test/e2e_node/runtimeclass_test.go

```diff
@@ -0,0 +1,149 @@
+
+		framework.Logf("command: %v: ", localCommand)
+		command += localCommand
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cgroup-verification-pod-" + string(uuid.NewUUID()),
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{
+				{
+					Image:   busyboxImage,
+					Name:    "container" + string(uuid.NewUUID()),
+					Command: []string{"sh", "-c", command},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "sysfscgroup",
+							MountPath: "/host_cgroups",
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: "sysfscgroup",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs/cgroup"},
+					},
+				},
+			},
+		},
+	}
+	return pod
+}
+
+var _ = framework.KubeDescribe("Kubelet PodOverhead handling [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("podoverhead-handling")
+	ginkgo.Describe("PodOverhead cgroup accounting", func() {
+		ginkgo.Context("On running pod with PodOverhead defined", func() {
+			ginkgo.It("Pod cgroup should be sum of overhead and resource limits", func() {
+				if !framework.TestContext.KubeletConfig.CgroupsPerQOS {
+					return
+				}
+
+				var (
+					guaranteedPod *v1.Pod
+					podUID        string
+					handler       string
+				)
+				ginkgo.By("Creating a RuntimeClass with Overhead definied", func() {
+					handler = e2enode.PreconfiguredRuntimeClassHandler(framework.TestContext.ContainerRuntime)
+					rc := &nodev1beta1.RuntimeClass{
+						ObjectMeta: metav1.ObjectMeta{Name: handler},
+						Handler:    handler,
+						Overhead: &nodev1beta1.Overhead{
+							PodFixed: getResourceList("200m", "140Mi"),
+						},
+					}
+					_, err := f.ClientSet.NodeV1beta1().RuntimeClasses().Create(context.TODO(), rc, metav1.CreateOptions{})
+					framework.ExpectNoError(err, "failed to create RuntimeClass resource")
+				})
+				ginkgo.By("Creating a Guaranteed pod with which has Overhead defined", func() {
+					guaranteedPod = f.PodClient().CreateSync(&v1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pod-with-overhead-" + string(uuid.NewUUID()),
+							Namespace: f.Namespace.Name,
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Image:     imageutils.GetPauseImageName(),
+									Name:      "container" + string(uuid.NewUUID()),
+									Resources: getResourceRequirements(getResourceList("100m", "100Mi"), getResourceList("100m", "100Mi")),
+								},
+							},
+							RuntimeClassName: &handler,
+							Overhead:         getResourceList("200m", "140Mi"),
+						},
+					})
+					podUID = string(guaranteedPod.UID)
+				})
+				ginkgo.By("Checking if the pod cgroup was created appropriately", func() {
+					cgroupsToVerify := []string{"pod" + podUID}
+					pod := makePodToVerifyCgroupSize(cgroupsToVerify, "30000", "251658240")
+					f.PodClient().Create(pod)
```

Comment:
I think podclient injects the namespace

```suggestion
					pod = f.PodClient().Create(pod)
```